pkg/common/secret/utils.go

```diff
@@ -0,0 +1,51 @@
+package secret
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GenerateGenericSecret(secretName string, secretNamespace string,
+	key string, value string) *corev1.Secret {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: secretNamespace,
+		},
+		Data: map[string][]byte{
+			key: []byte(value),
+		},
+		Type: "Opaque",
+	}
+
+	return secret
+}
+
+func DeleteSecret(secretName string, secretNamespace string,
+	ctx context.Context, providedClient client.Client) error {
+	objectKey := client.ObjectKey{
+		Name:      secretName,
+		Namespace: secretNamespace,
+	}
+	current := &corev1.Secret{}
+	// Get the current secret if it exists.
+	err := providedClient.Get(ctx, objectKey, current)
```

Comment:
I've seen upstream that you can simplify the delete by removing the Get like so:

```go
if err := client.Delete(ctx, &obj); client.IgnoreNotFound(err) != nil {
  return errors.WithStack(err)
}
return nil
```

Reply:
Then maybe we do not need a util function to replace this one liner :)

Reply:
Agreed, there's no need to get in order to delete a resource.

Reply:
I simplified the deletion by getting rid of the get but I couldn't go as far as the one-liner example that you gave :/ 
Also I kept the secret deletion function to avoid code duplication.

WDYT ?

Reply:
That's fine with me